Overhaul "Install" instructions (#48)
gbdev/rgbds-www: src/pages/install/index.md

# Installing RGBDS

Most people will want to use a release (the more recent the better), but if you are willing to help us test new features, you may consider [using a development version](/install/master).

The two main options to install a release of RGBDS are to download pre-built executables, or to [build from source](/install/source).
Below is a list of <abbr title="Operating System">OS</abbr>es for which we know pre-built executables are available.

If your OS isn’t listed below, try finding `rgbds` in your package manager—refer to your OS' documentation for help.
If you still can’t find RGBDS, or the specific version you are looking for is unavailable, you will have to [compile it from source](#building-from-source).

<Tabs>
<TabItem value="linux" label="Linux">

> TO DO

Reviewer: We should suggest, in order:

- Using the distribution's package manager, if available (providing some examples?)
- Using our pre-built binaries
- Docker?
- Compiling from source
- ??? Anything I missed?

</TabItem>

<TabItem value="windows" label="Windows">

The install instructions change a bit depending on the environment you wish to use RGBDS with.

<details>

> <summary>"Plain" Windows (CMD, PowerShell)</summary>

Reviewer: Actually, the WSL instructions should instruct to refer to the Linux instructions for the corresponding Linux distribution (noting that if you don't know, it's probably Ubuntu).

I'm thinking we should organise the tabs something like `WSL`, then `MSYS2 or Cygwin`, and finally `none of these`. That should make what the last one is about, clearer. I think.

Author: This seems a good idea, although I have no experience whatsoever on Windows so if you have any specific ideas on how to reorganize the Windows tab feel free to push

Reviewer: I'm not a Windows user anymore, but I think I have some ideas. I'll do that then.

Reviewer: OK, done!

# Installing RGBDS

Most people will want to use a stable release (the more recent the better). 

> Below is a list of Operative Systems for which pre-built executables are available. If none of these options fits your needs, you can [build from source](/install/source).
> 

Reviewer: ```suggestion
Below is a list of OSes for which pre-built executables are available.
If none of these options fits your needs, you can [build from source](./source.md).
```
We can use `<abbr title="Operating Systems">` to explain "OSes" instead, if desired.
Additionally, linking to the Markdown file makes it checked by Docusaurus, which improves reliability.

Author: Agree!

<Tabs>
<TabItem value="linux" label="Linux">

Linux x86_64 ready binaries are available in our [Releases](https://github.com/gbdev/rgbds/releases) page.

> ##### Arch

Reviewer: ```suggestion
## Arch
```
We're only at heading level 2 here; level 3 might be appropriate too.

RGBDS is available in the official Arch Linux repositories as [`community/rgbds`](https://www.archlinux.org/packages/community/x86_64/rgbds/); you can also get the [latest master](/docs/master) via the [`rgbds-git`](https://aur.archlinux.org/packages/rgbds-git) AUR package.

```bash
pacman -S rgbds
```

> 

Reviewer: ```suggestion

```

</TabItem>

> <TabItem value="arch" label="Arch Linux">

Reviewer: Let's move all distro-specific instructions to the "Linux" tab, right?

</details>

</TabItem>

<TabItem value="docker" label="Docker">

We distribute an [official container image for RGBDS](https://github.com/gbdev/rgbds/pkgs/container/rgbds). It contains the built executables and the build dependencies in case you want to compile from source.

```bash
docker pull ghcr.io/gbdev/rgbds:latest
```

</TabItem>
</Tabs>

<br />

> ### Installing a development version

Reviewer: ```suggestion
## Installing a development version
```

> If you are willing to help us test new features, consider [using a development version](/install/master).

Reviewer: ```suggestion
If you are willing to help us test new features, consider [using a development version](./master.md).
```

> ### Managing multiple versions

Reviewer: ```suggestion
## Managing multiple versions
```
